service-front/module/Application/src/View/Helper/RouteName.php

```diff
@@ -2,32 +2,29 @@
 
 namespace Application\View\Helper;
 
-use Laminas\Router\RouteMatch;
 use Laminas\View\Helper\AbstractHelper;
 
 class RouteName extends AbstractHelper
 {
     /**
-     * @var RouteMatch
+     * @var array
      */
-    private $routeMatch;
+    private $routes;
 
     /**
-     * @param RouteMatch $routeMatch
+     * @param string|null $currentRoute
+     * @param string|null $previousRoute
      */
-    public function __construct(?RouteMatch $routeMatch)
+    public function __construct(?string $currentRoute, ?string $previousRoute)
     {
-        $this->routeMatch = $routeMatch;
+        $this->routes = [
+            'current'   => $currentRoute,
+            'previous'  => $previousRoute
+        ];
     }
 
     public function __invoke()
     {
-        $routeName = '';
-
-        if ($this->routeMatch) {
-            $routeName = $this->routeMatch->getMatchedRouteName();
-        }
-
-        return $routeName;
+        return $this->routes;
```

Comment:
Just out of interest, is `__invoke()` ever called? I imagine instances of this will come from `RouteNameFactory`, which calls the constructor for this class with `new`.

Reply:
It's invoked when `RouteName` is used within a twig template.